front/app/containers/IdeasNewSurveyPage/IdeasNewSurveyForm/index.tsx

```diff
@@ -188,10 +190,23 @@ const IdeasNewSurveyForm = ({ project, phaseId }: Props) => {
       publication_status: data.publication_status || 'published',
     };
 
+    const handleOnError = () => {
+      // If an error happens, it's likely some permission issues.
+      // We refetch the project to use the correct action descriptors.
+      queryClient.invalidateQueries({
+        queryKey: projectsKeys.all(),
```

Comment:
`projectsKeys.item({ id: project.data.id })` didn't invalidate the correct cache because this project is fetched by `slug`, not by `id`. So, to make it less brittle (in case we change the way we fetch the project), I invalidate all cache keys. Anyway, it should not happen often.

Reply:
Indeed it worked, but I assumed it would be more brittle than `projectsKeys.all()`. E.g., what if we'll change it to fetching this project by id, not by slug in the future? This code will break and there's no way it'll be noticed because the bug is pretty rare. So, I decided to make it less precise, but more reliable.

Reply:
> we invalidate all projects and a new api call is made for all projects

I see your point, but I'm not sure it's a problem. I believe it's really rare when a user submits a survey a second time when it's not allowed because our UI doesn't allow it normally (the button is blocked).